File: content/LLD/Reference-Code/22. Ride Sharing Project - Good Example/location.py
from math import sqrt


class Location:
    def __init__(self, lat: float, long: float):
        self.__lat: float = lat
        self.__long: float = long

    def get_latitude(self) -> float:
        return self.__lat

    def get_longitude(self) -> float:
        return self.__long

    def calcDistance(self, loc: Location):
        # Euclidean Distance
        dx: float = self.get_latitude() - loc.get_latitude()
        dy: float = self.get_longitude() - loc.get_longitude()
        return sqrt(dx * dx + dy * dy)

File: content/LLD/Reference-Code/22. Ride Sharing Project - Good Example/Location.java
public class Location {
    private final double lat;
    private final double lon;

    public Location(double lat, double lon) {
        this.lat = lat;
        this.lon = lon;
    }

    public double getLatitude() {
        return this.lat;
    }

    public double getLongitude() {
        return this.lon;
    }

    public double calcDistance(Location loc) {
        double dx = this.getLatitude() - loc.getLatitude();
        double dy = this.getLongitude() - loc.getLongitude();
        return Math.sqrt(dx * dx + dy * dy);
    }
}
